### PR TITLE
implement login by credentials and return jwt token

### PR DESCRIPTION
**Pull request by ErfanMojibi**

### CODE REVIEW

File: src/main/kotlin/ir/jaamebaade/jaamebaade/service/AuthenticationService.kt [hunk 1]
```diff
@@ -28,4 +39,14 @@ class AuthenticationService(
         return user.id
 
     }
+    fun login() : UserDetails? {
```
Comment (MuhammadKhosravi) [inline src/main/kotlin/ir/jaamebaade/jaamebaade/service/AuthenticationService.kt:42]: why do we need to implement this function?

File: src/main/kotlin/ir/jaamebaade/jaamebaade/controller/AuthenticationController.kt [hunk 2]
```diff
@@ -21,4 +22,11 @@ class AuthenticationController(
             ?: return BaseResultFactory.badRequest("User already exists")
         return BaseResultFactory.ok(id)
     }
+
+    @PostMapping(value = ["/login"], consumes = ["application/json"], produces = ["application/json"])
```
Comment (MuhammadKhosravi) [inline src/main/kotlin/ir/jaamebaade/jaamebaade/controller/AuthenticationController.kt:26]: function name is `register` but it should be `login`

File: src/main/kotlin/ir/jaamebaade/jaamebaade/service/AuthenticationService.kt [hunk 3]
```diff
@@ -28,4 +39,14 @@ class AuthenticationService(
...
+        TODO()
+    }
+    fun loginByCredentials(loginUserRequest: LoginUserRequest): String? {
```
Comment (MuhammadKhosravi) [inline src/main/kotlin/ir/jaamebaade/jaamebaade/service/AuthenticationService.kt:45]: why not simply login?

File: src/main/kotlin/ir/jaamebaade/jaamebaade/controller/AuthenticationController.kt [hunk 4]
```diff
@@ -21,4 +22,11 @@ class AuthenticationController(
...
+    fun register(@RequestBody loginUserRequest: LoginUserRequest): ResponseEntity<BaseResult> {
```
Comment (MuhammadKhosravi) [inline src/main/kotlin/ir/jaamebaade/jaamebaade/controller/AuthenticationController.kt:27]: function name is `register` but it should be `login`

File: src/main/kotlin/ir/jaamebaade/jaamebaade/service/AuthenticationService.kt (hunk 1 above)
Comment (ErfanMojibi) [reply src/main/kotlin/ir/jaamebaade/jaamebaade/service/AuthenticationService.kt:42]: Login: for login with token
LoginByCredentials: for login by username and password?

**Pull request merged**